File: src/main/java/gargoyle/ct/ui/CTControlActions.java
package gargoyle.ct.ui;

import gargoyle.ct.config.CTConfig;
import gargoyle.ct.config.CTConfigs;
import org.jetbrains.annotations.Nullable;

public interface CTControlActions {
    void about();

    void arm(CTConfig config);

    void browseConfigs();

    void exit();

    void help();

    CTConfigs loadConfigs(boolean reload);

    void saveConfigs(CTConfigs configs);

    @Nullable CTConfig showNewConfig();

    void showPreferences();

    void unarm();
}

File: src/main/kotlin/gargoyle/ct/ui/CTControlActions.kt
package gargoyle.ct.ui

import gargoyle.ct.config.CTConfig
import gargoyle.ct.config.CTConfigs

interface CTControlActions {
    fun about()
    fun arm(config: CTConfig)
    fun browseConfigs()
    fun exit()
    fun help()
    fun loadConfigs(reload: Boolean): CTConfigs
    fun saveConfigs(configs: CTConfigs)
    fun showNewConfig(): CTConfig?
    fun showPreferences()
    fun unarm()
}
